app/Http/Controllers/AnnounceController.php

```diff
@@ -48,446 +66,443 @@ class AnnounceController extends Controller
-        }
-
-        // Request Agent Information
-        $agent = $request->server('HTTP_USER_AGENT');
-
-        // Blacklist
-        if (\config('client-blacklist.enabled') == true) {
-            // Check If Browser Is Blacklisted
-            $blockedBrowsers = \config('client-blacklist.browsers');
-            if (\in_array($agent, $blockedBrowsers)) {
-                \abort(405, 'What Are You Trying To Do?');
-                exit();
-            }
-
-            // Check If Client Is Blacklisted
-            $blockedClients = \config('client-blacklist.clients');
-            if (\in_array($agent, $blockedClients)) {
-                //info('Blacklist Client Attempted To Connect To Announce');
-                return \response(Bencode::bencode(['failure reason' => 'The Client You Are Trying To Use Has Been Blacklisted']))->withHeaders(['Content-Type' => 'text/plain']);
-            }
+        try {
+            /**
+             * Check client.
+             */
+            $this->checkClient($request);
+
+            /**
+             * Check passkey.
+             */
+            $this->checkPasskey($passkey);
+
+            /**
+             * Check user via supplied passkey.
+             */
+            $user = $this->checkUser($passkey);
+
+            /**
+             * Check and then get Announce queries.
+             */
+            $queries = $this->checkAnnounceFields($request);
+
+            /**
+             * Get Torrent Info Array from queries and judge if user can reach it.
+             */
+            $torrent = $this->checkTorrent($queries['info_hash']);
+
+            /**
+             * Lock Min Announce Interval.
+             */
+            $this->checkMinInterval($queries, $user);
+
+            /**
+             * Check Download Slots.
+             */
+            //$this->checkDownloadSlots($user);
+
+            /**
+             * Generate A Response For The Torrent Clent.
+             */
+            $rep_dict = $this->generateSuccessAnnounceResponse($queries, $torrent, $user);
+
+            /**
+             * Dispatch The Specfic Annnounce Event Job.
+             */
+            $this->sendAnnounceJob($queries, $user, $torrent);
+        } catch (TrackerException $exception) {
+            $rep_dict = $this->generateFailedAnnounceResponse($exception);
+        } finally {
+            return $this->sendFinalAnnounceResponse($rep_dict);
         }
+    }
 
-        // If Passkey Is Not Provided Return Error to Client
-        if ($passkey == null) {
-            //info('Client Attempted To Connect To Announce Without A Passkey');
-            return \response(Bencode::bencode(['failure reason' => 'Please Call Passkey']))->withHeaders(['Content-Type' => 'text/plain']);
+    /**
+     * @param Request $request
+     *
+     * @throws \App\Exceptions\TrackerException
+     *
+     * @return void
+     */
+    protected function checkClient(Request $request): void
```

Comment:
So did some logging and am seeing so far that these two torrent versions are sending the `accept-language` header.

Client Headers: ["uTorrent/15399 CFNetwork/978.2 Darwin/18.7.0 (x86_64)","en-us",null,null,null]	
Client Headers: ["uTorrent/3220(28500)","en-CA",null,null,null]